src/solver/pooling/backward2d.cpp

```diff
@@ -0,0 +1,216 @@
+}
+
+ConvSolution
+PoolingBackward2d::GetSolution(const ExecutionContext&,
+                               const miopen::pooling::ProblemDescription& problem) const
+{
+    auto result = ConvSolution{miopenStatusSuccess};
+
+    {
+        auto kernel = KernelInfo{};
+
+        kernel.kernel_file = "MIOpenPoolingBwd.cl";
+
+        if(problem.GetPooling().GetMode() == miopenPoolingMax)
+        {
+            kernel.kernel_name = "mloPoolingMaxBwd";
+        }
+        else if(problem.GetPooling().GetMode() == miopenPoolingAverage ||
+                problem.GetPooling().GetMode() == miopenPoolingAverageInclusive)
+        {
+            kernel.kernel_name = "mloPoolingAveBwd";
+        }
+
+        std::size_t batch_sz, n_inputs, in_height, in_width;
+        std::tie(batch_sz, n_inputs, in_height, in_width) =
+            miopen::tien<4>(problem.GetXDesc().GetLengths(), 1);
+
+        const int pooling_method = (problem.GetPooling().GetMode() == miopenPoolingMax)
+                                       ? MLO_POOLING_OP_MAX
+                                       : ((problem.GetPooling().GetMode() == miopenPoolingAverage)
+                                              ? MLO_POOLING_OP_AVE
+                                              : MLO_POOLING_OP_AVE_INCLUSIVE);
+
+        std::size_t grp_tile0 = 8;
+        std::size_t grp_tile1 = 8;
+
+        std::size_t out_pix_tile0 = 1;
+        std::size_t out_pix_tile1 = 1;
+
+        if(problem.GetPooling().GetMode() == miopenPoolingMax)
+        {
+            // clang-format off
+            grp_tile0 =
+                   in_width <= 8   ? 8
+                : (in_width <= 16  ? 4
+                : (in_width <= 24  ? 8
+                : (in_width <= 32  ? 32
+                : (in_width <= 64  ? 8
+                : (in_width <= 96  ? 16
+                : (in_width <= 128 ? 16
+                                   : 32))))));
+            grp_tile1 =
+                   in_width <= 8   ? 8
+                : (in_width <= 16  ? 16
+                : (in_width <= 24  ? 8
+                : (in_width <= 32  ? 4
+                : (in_width <= 64  ? 8
+                : (in_width <= 96  ? 4
+                : (in_width <= 128 ? 16
+                                   : 4))))));
+            // clang-format on
+
+            out_pix_tile0 = in_width > 8 && in_width <= 24 ? 4 : 1;
+            out_pix_tile1 = in_width <= 24 ? 1 : (in_width > 64 && in_width <= 96 ? 4 : 8);
+        }
+
+        int g_wk_width = ((in_width + grp_tile0 * out_pix_tile0 - 1) / (grp_tile0 * out_pix_tile0));
+        int g_wk_height =
+            ((in_height + grp_tile1 * out_pix_tile1 - 1) / (grp_tile1 * out_pix_tile1));
+
+        const auto kernel_size_w   = problem.GetPooling().lens[0];
+        const auto kernel_size_h   = problem.GetPooling().lens[1];
+        const auto kernel_stride_w = problem.GetPooling().strides[0];
+        const auto kernel_stride_h = problem.GetPooling().strides[1];
```

Comment:
@DrizztDoUrden It seems like we need to swap `_h` and `_w` here to fix 2d pooling.